api/swagger.yml

```
@@ -1,507 +1,795 @@
-        $ref: "#/definitions/staging_location"
-      checksum:
-        type: string
-        description: unique identifier of object content on backing store (typically ETag)
-      mtime:
-        type: integer
-        format: int64
-      size_bytes:
-        type: integer
-        format: int64
-    required:
-      - staging
-      - checksum
-      - mtime
-      - size_bytes
-
-  object_stage_creation:
-    type: object
-    required:
-      - physical_address
-      - checksum
-      - size_bytes
-    properties:
-      physical_address:
-        type: string
-      checksum:
-        type: string
-      size_bytes:
-        type: integer
-        format: int64
-      metadata:
-        type: object
-        additionalProperties:
+    RepositoryCreation:
+      type: object
+      required:
+        - name
+        - storage_namespace
+      properties:
+        name:
+          type: string
+          pattern: '^[a-z0-9][a-z0-9-]{2,62}$'
+        storage_namespace:
+          type: string
+          description: "Filesystem URI to store the underlying data in (e.g. 's3://my-bucket/some/path/')"
+          example: "s3://example-bucket/"
+          pattern: '^(s3|gs|https?|mem|local|transient)://.*$'
+        default_branch:
+          example: "master"
           type: string
 
-  underlying_object_properties:
-    type: object
-    properties:
-      storage_class:
-        x-nullable: true
-        type: string
+    ObjectStats:
+      type: object
+      required:
+        - checksum
+        - physical_address
+        - path
+        - path_type
+        - mtime
+      properties:
+        path:
+          type: string
+        path_type:
+          type: string
+          enum: [ common_prefix, object ]
+        physical_address:
+          type: string
+        checksum:
+          type: string
+        mtime:
+          type: integer
+          format: int64
+        size_bytes:
+          type: integer
+          format: int64
+
+    ObjectStatsList:
+      type: object
+      required:
+        - pagination
+        - results
+      properties:
+        pagination:
+          $ref: "#/components/schemas/Pagination"
+        results:
+          type: array
+          items:
+            $ref: "#/components/schemas/ObjectStats"
+
+    ObjectStageCreation:
+      type: object
+      required:
+        - physical_address
+        - checksum
+        - size_bytes
+      properties:
+        physical_address:
+          type: string
+        checksum:
+          type: string
+        size_bytes:
+          type: integer
+          format: int64
+        metadata:
+          type: object
+          additionalProperties:
+            type: string
 
-  ref:
-    type: object
-    required:
-      - id
-      - commit_id
-    properties:
-      id:
-        type: string
-      commit_id:
-        type: string
+    UnderlyingObjectProperties:
+      type: object
+      properties:
+        storage_class:
+          x-nullable: true
```

I think this was a Swaggerism.  Is this still TWTDI in OpenAPI 3?  I _think_ I added this because Swagger used OpenAPI 2 but go-swagger couldn't distinguish between a missing and an empty value without it.  Here I believe you might be able to say:
```suggestion
          nullable: true
```

Or just (or additionally) allow it to drop outright, and let our New Swagger Code Generator DTRT and distinguish between missing and empty.

---

guess the openapi2 to 3 didn't catch it - updated to nullable.